tests/core/matrix.rs

```diff
@@ -1263,6 +1263,20 @@ fn column_iterator_double_ended_mut() {
     assert_eq!(col_iter_mut.next(), None);
 }
 
+#[test]
+fn test_inversion_failure_leaves_matrix_unchanged() {
```

Comment:
Maybe rename to `_matrix4_unchanged` to indicate that this was specifically an issue with 4x4 inversion and what we're testing here